lib/bolt/inventory.rb

```diff
@@ -174,7 +174,7 @@ def update_target(target)
         data['config'] = {}
       end
 
-      unless Bolt::Util.windows? || data['config']['transport']
+      unless data['config']['transport']
         data['config']['transport'] = 'local' if target.name == 'localhost'
```

Comment:
Whole thing could be collapsed to `data['config']['transport'] ||= 'local' if target.name == 'localhost'`.